Restapi 1487 dotnet file transfer examples

iB0nes opened the pull request:
Added examples for upload and download for ops and transfer endpoints.
Added documentation README
Updated arch/external-storage with `max_part_size` explanation.
`max_part_size` set by default to 1GB.

jpdorsch commented on docs/user_guide/data_transfer_dotnet/README.md:
````diff
@@ -0,0 +1,124 @@
+This directory includes a collection of classes representing structured data types, used for serializing and deserializing requests and responses exchanged with FirecREST.
+### Endpoints
+This directory holds all the classes required to interact with FirecREST API endpoints, designed following an inheritance-based architecture. The following class diagram represents the structure of them.
+```mermaid
````
Please, use the [mkdocs-material](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) code example format: `!!! example` or `??? example` for expanded/hidden content.
For instance

```
!!! example "Endpoints definition"
    ```mermaid
    (...)
    ```
```

iB0nes replied:
Hi, "mermaid" is not a code example: it is a schematic diagram, it should appear as a picture. If I use the !!! notation, the schematic is no more visible.
However I added it around all code blocks as you requested, but the result I'm seeing opening the README.md inside GitHub is this in the attached screenshot.
I see the same in other READMEs in the project even with Python examples.
Is it intended to be render that in some way, or am I missing someting?
Thanks.

![image](https://github.com/user-attachments/assets/26494449-e091-442b-92dd-f359478252a3)

jpdorsch replied:
Yes, you'll see the README.md preview like that.
To see it correctly with the mkdocs format, you should create a `venv`, install requirements and launch the python HTTP server

```
$ python -m venv .venv
$ source .venv/bin/activate
$ (.venv) pip install -r docs/requirements.txt
$ (.venv) make clean
$ (.venv) make docs
$ (.venv) cd site
$ (.venv) python -m http.server 8001
```

and then use your browser to navigate to HTTP://localhost:8001

iB0nes replied:
Thank you, now it's crystal clear.
I replaced the mermaid diagram with a real one (it's better readable) and I added a link from the user_guide's README.
It should be better now.

jpdorsch commented on docs/user_guide/data_transfer_dotnet/README.md:
````diff
@@ -0,0 +1,124 @@
+## Examples
+### Example 1: base authentication 
+The [Program.cs](firecrest_base/Program.cs) in the [firecrest_base](firecrest_base/) project demonstrates the simplest connection to FirecREST. It initializes the [EndpointStatus](firecrest_base/Endpoints/EndpointStatus.cs) class and retrieves the Systems API result.
+```
````
Same as above

jpdorsch commented on docs/user_guide/data_transfer_dotnet/README.md:
````diff
@@ -0,0 +1,124 @@
+
+### Example 2: large files transfer
+The [Program.cs](large_files_transfer/Program.cs) in the [large_files_transfer](large_files_transfer) project generates a random content file with an arbitrary customizable by the function:
+```
````
Same as above

iB0nes replied:
done

jpdorsch reviewed (changes requested):
Just small format changes.

jpdorsch commented on docs/user_guide/data_transfer_dotnet/README.md:
```diff
@@ -0,0 +1,135 @@
+## General description
+This .NET solution provides examples demonstrating how to connect to FirecREST and transfer files using C# via different API endpoints. The implementation follows a fully `async` approach.
+
+The Examples do not requires parameters. You may be able to build them and run the generated executable. All of them, by default, target the [Docker-Compose demo environment](../../../README.md#running-firecrest-v2-with-docker-compose).
```
Change the reference on this link: `[Docker-Compose demo environment](../../getting_started/#trying-firecrest-in-a-containerised-environment)`

iB0nes replied:
done

jpdorsch commented on docs/setup/arch/external_storage/README.md:
```diff
@@ -12,7 +12,7 @@ The S3 storage can be either on-premises or cloud-based. In any case it is requi
 Uploading data from the extern of the HPC infrastructure requires users to apply the [multipart upload protocol](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html), therefore the data file shall be divided into parts. The size limit of each part is defined in the response of FirecREST to the upload call. 
 
-Tha maximum size of the parts can be configured via the [`max_part_size parameter`](../../../setup/conf/README.md#bucket_lifecycle_configuration). The specified value must be lower than [S3's maximum limit](https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html) of 5GB per part and it is applied to any upload data call.
+Tha maximum size of the parts can be configured via the [`max_part_size parameter`](../../../setup/conf/README.md#bucket_lifecycle_configuration). The specified value must be lower than [S3's maximum limit](https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html) of 5GB per part and it is applied to any upload data call. When setting this value, please consider that clients using this p ...
```
2 links should change in this file:
- `[bucket_lifecycle_configuration](../../../setup/conf/README.md#bucketlifestyleconfiguration)`
- `[max_part_size parameter](../../../setup/conf/README.md#multipartupload)`

jpdorsch reviewed (changes requested):
I found some links that are not working, I have pointed you the fixed links

iB0nes commented on docs/user_guide/data_transfer_dotnet/README.md:
````diff
@@ -0,0 +1,124 @@
+
+### Example 3: small files transfer
+The [Program.cs](small_files_transfer/Program.cs) in the [small_files_transfer] (small_files_transfer) project generates a random content file with an arbitrary customizable size using a similar function as implemented in Example 2. In this case, the size is expressed in kilobytes (KB). Please note that the maximum file size allowed for direct upload and download using the `ops` endpoint is 1MB.  ...
+```
````
done

iB0nes commented:
Fixed navigation menu in mkdocs.yaml.
I found some warnings about links not working properly with `make docs`. I fixed them.
Since the name of the directory with the .NET code was not matching links and titles, I renamed it (I can't rename the Solution, but at least the directory is correct).

jpdorsch reviewed (approved):
LGTM, thanks!
